### PR TITLE
[DON'T MERGE]: Upcoming HttpClient

### PR DESCRIPTION
**Pull request by baulig**

### CODE REVIEW

File: profiles/monodroid/System.Net.Http.cs [hunk 1]
```diff
@@ -19,6 +19,7 @@
 [assembly:System.Runtime.CompilerServices.InternalsVisibleToAttribute("System.Net.Http.WebRequest, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
 [assembly:System.Runtime.CompilerServices.RuntimeCompatibilityAttribute(WrapNonExceptionThrows=true)]
 [assembly:System.Runtime.InteropServices.ComVisibleAttribute(false)]
+[assembly:System.Security.Permissions.SecurityPermissionAttribute(System.Security.Permissions.SecurityAction.RequestMinimum, SkipVerification=true)]
```
Comment (filipnavara) [inline profiles/monodroid/System.Net.Http.cs]: Does this come from CoreFX? Generally speaking it's obsolete and should be removed. I thought it was already done globally on the CoreFX side.

File: profiles/net_4_x/System.Net.Http.WebRequest.cs [hunk 2]
```diff
@@ -61,13 +61,12 @@ public partial class WebRequestHandler : System.Net.Http.HttpClientHandler
         public bool AllowPipelining { get { throw null; } set { } }
         public System.Net.Security.AuthenticationLevel AuthenticationLevel { get { throw null; } set { } }
         public System.Net.Cache.RequestCachePolicy CachePolicy { get { throw null; } set { } }
-        public new System.Security.Cryptography.X509Certificates.X509CertificateCollection ClientCertificates { get { throw null; } }
         [System.MonoTODOAttribute]
         public System.TimeSpan ContinueTimeout { get { throw null; } set { } }
         public System.Security.Principal.TokenImpersonationLevel ImpersonationLevel { get { throw null; } set { } }
-        public new int MaxResponseHeadersLength { get { throw null; } set { } }
```
Comment (filipnavara) [inline profiles/net_4_x/System.Net.Http.WebRequest.cs]: I could be wrong, but these two APIs are still present in .NET Framework 4.7.2. Wouldn't it cause errors for code that was compiled against the framework and uses these APIs? If yes, can you add these properties back and implement them by calling the `base.` properties that they override?

File: profiles/monodroid/System.Net.Http.cs (hunk 1 above)
Comment (baulig) [reply profiles/monodroid/System.Net.Http.cs]: It's certainly possible.  I should probably rerun the api snapshot tool to make sure we're actually using the latest version.

File: profiles/net_4_x/System.Net.Http.WebRequest.cs (hunk 2 above)
Comment (filipnavara) [reply profiles/net_4_x/System.Net.Http.WebRequest.cs]: I gave it a try and removing the "new" method actually works, so you can disregard the comment.

**Pull request closed without merging**